fastlane/lib/fastlane/actions/automatic_code_signing.rb

```diff
@@ -40,6 +40,12 @@ def self.run(params)
           end
           if params[:code_sign_identity]
             build_configuration_list.set_setting("CODE_SIGN_IDENTITY", params[:code_sign_identity])
+
+            # We also need to update the value if it was overriden for a specific SDK
+            build_configuration_list.build_configurations.each do |build_configuration|
+              codesign_build_settings_keys = build_configuration.build_settings.keys.select { |key| key.to_s.match(/CODE_SIGN_IDENTITY.*/) }
+              codesign_build_settings_keys.map { |key| build_configuration_list.set_setting(key, params[:code_sign_identity]) }
```

Comment:
Can you change this `map` to an `each`? This doesn't need to be map since nothing is being returned or modified on `codesign_build_settings_keys` 🙃